### PR TITLE
sessionctx: Fix SET GLOBAL tidb_skip_isolation_level_check=1

### LINKED ISSUE #27897: SET GLOBAL tidb_skip_isolation_level_check=1 doesn't work as expected
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> SET GLOBAL tidb_skip_isolation_level_check=1;
Query OK, 0 rows affected (0.02 sec)
```
Disconnect and reconnect to TiDB
```
mysql> SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;
ERROR 8048 (HY000): The isolation level 'SERIALIZABLE' is not supported. Set tidb_skip_isolation_level_check=1 to skip this error
```

### 2. What did you expect to see? (Required)

```
mysql> SET GLOBAL tidb_skip_isolation_level_check=1;
Query OK, 0 rows affected (0.03 sec)
```
Disconnect and reconnect to TiDB
```
mysql> SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;
Query OK, 0 rows affected, 1 warning (0.00 sec)
```

### 3. What did you see instead (Required)

```
mysql> SET GLOBAL tidb_skip_isolation_level_check=1;
Query OK, 0 rows affected (0.02 sec)
```
Disconnect and reconnect to TiDB
```
mysql> SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;
ERROR 8048 (HY000): The isolation level 'SERIALIZABLE' is not supported. Set tidb_skip_isolation_level_check=1 to skip this error
```

### 4. What is your TiDB version? (Required)

5.1.0 and all the new versions

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Fix issue that SET GLOBAL tidb_skip_isolation_level_check=1 doesn't work as isolation level check code looks for session variable only.

Signed-off-by: Xiaoguang Sun <sunxiaoguang@pingcap.com>

### What problem does this PR solve?

Issue Number: close #27897 

Problem Summary:

### What is changed and how it works?

What's Changed:
Checking isolation code looks for tidb_skip_isolation_level_check from session variables only which makes it behaves differently then it is in earlier version than 5.1.0

How it Works:
tidb_skip_isolation_level_check was marked as skipInit accidentally, removing it fixes this issue. 

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

```
mysql> SET GLOBAL tidb_skip_isolation_level_check=1;
Query OK, 0 rows affected (0.03 sec)

mysql> SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;
Query OK, 0 rows affected, 1 warning (0.00 sec)
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/varsutil.go [hunk 1]
```diff
@@ -152,7 +152,11 @@ func checkReadOnly(vars *SessionVars, normalizedValue string, originalValue stri
 func checkIsolationLevel(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
 	if normalizedValue == "SERIALIZABLE" || normalizedValue == "READ-UNCOMMITTED" {
 		returnErr := ErrUnsupportedIsolationLevel.GenWithStackByArgs(normalizedValue)
-		if !TiDBOptOn(vars.systems[TiDBSkipIsolationLevelCheck]) {
+		value, err := GetSessionOrGlobalSystemVar(vars, TiDBSkipIsolationLevelCheck)
+		if err != nil {
+			return normalizedValue, err
+		}
+		if !TiDBOptOn(value) {
```
Comment (morgo) [inline sessionctx/variable/varsutil.go:159]: Shouldn't this match the corresponding scope of what you are trying to change? i.e. to change the global isolation level, it checks the global scoped skip setting.

This is the behavior of `checkReadOnly`:

https://github.com/pingcap/tidb/blob/dfd358a9882b93195305ba2e9f31e0c89a058290/sessionctx/variable/varsutil.go#L131-L150

**Comment by morgo** [comment]
The root cause is because `TiDBSkipIsolationLevelCheck` is set to `skipInit: true`:

https://github.com/pingcap/tidb/blob/7dbf7b31c526c5d1819bcfea8ce00e66ce0af8a9/sessionctx/variable/sysvar.go#L1272

This was done in https://github.com/pingcap/tidb/pull/24836 ; where the previous compatibility was ensured as part of a refactor, but in this case `skipInit` is not the correct behavior.

(For an example where it is correct, see "instance scoped" variables. This is a case you don't want the SetSession function called on session init).

**Comment by sunxiaoguang** [comment]
> The root cause is because `TiDBSkipIsolationLevelCheck` is set to `skipInit: true`:
> 
> https://github.com/pingcap/tidb/blob/7dbf7b31c526c5d1819bcfea8ce00e66ce0af8a9/sessionctx/variable/sysvar.go#L1272
> 
> This was done in #24836 ; where the previous compatibility was ensured as part of a refactor, but in this case `skipInit` is not the correct behavior.
> 
> (For an example where it is correct, see "instance scoped" variables. This is a case you don't want the SetSession function called on session init).

Cool, that makes the fix even simpler. Let me try this way tomorrow. Thanks

**Comment by sunxiaoguang** [comment]
> > The root cause is because `TiDBSkipIsolationLevelCheck` is set to `skipInit: true`:
> > https://github.com/pingcap/tidb/blob/7dbf7b31c526c5d1819bcfea8ce00e66ce0af8a9/sessionctx/variable/sysvar.go#L1272
> > 
> > This was done in #24836 ; where the previous compatibility was ensured as part of a refactor, but in this case `skipInit` is not the correct behavior.
> > (For an example where it is correct, see "instance scoped" variables. This is a case you don't want the SetSession function called on session init).
> 
> Cool, that makes the fix even simpler. Let me try this way tomorrow. Thanks

Remove `skipInit` works well. PTAL @morgo, thanks.

**Review by morgo — APPROVED** [review]

**Comment by sunxiaoguang** [comment]
I will add unit test to avoid regression in future.

**Comment by sunxiaoguang** [comment]
Unstable tests failing were resolved, @bb7133 PTAL, Thanks.

**Review by bb7133 — APPROVED** [review]

**Comment by sunxiaoguang** [comment]
> @sunxiaoguang: `/merge` is only allowed for the committers, you can assign this pull request to the committer in [list](https://prow.tidb.io/tichi/repos/pingcap/tidb/pulls/27898/owners) by filling `/assign @committer` in the comment to help merge this pull request.

Conflict fixed, PTAL @morgo , Thanks.

**Review by morgo — APPROVED** [review]

**Pull request merged**